src/openzaak/components/documenten/tasks.py

```diff
@@ -0,0 +1,377 @@
+
+    if document_row.uuid:
+        try:
+            uuid = UUID(document_row.uuid, version=4)
+        except ValueError:
+            error_message = (
+                f"Given UUID for row {row_index} is not a valid UUID (version 4)"
+            )
+
+            logger.warning(error_message)
+            document_row.comment = error_message
+            document_row.processed = True
+
+            return document_row
+
+        if document_row.uuid in existing_uuids:
+            error_message = (
+                f"UUID given on row {row_index} was already found! Not overwriting "
+                "existing EIO."
+            )
+
+            logger.warning(error_message)
+            document_row.comment = error_message
+            document_row.processed = True
+
+            return document_row
+
+        data["uuid"] = str(uuid)
+    else:
+        data["uuid"] = str(uuid4())
+
+    gegevensgroep_fields = ("ondertekening", "integriteit")
+
+    for field in gegevensgroep_fields:
+        gegevens_groep_value = data.pop(field, {})
+
+        if not gegevens_groep_value:
+            continue
+
+        for key, value in gegevens_groep_value.items():
+            data[f"{field}_{key}"] = value
+
+    instance = EnkelvoudigInformatieObject(**data)
+    instance.canonical = EnkelvoudigInformatieObjectCanonical()
+
+    if not instance.identificatie:
+        instance.identificatie = generate_unique_identification(
+            instance, "creatiedatum"
+        )
+
+    zaak_id = document_row.zaak_id
+
+    if zaak_id and zaak_id not in zaak_uuids:
+        error_message = f"Zaak ID specified for row {row_index} is unknown."
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    try:
+        instance.clean()
+    except ValidationError as e:
+        error_message = (
+            "A validation error occurred while validating a "
+            f"EnkelvoudigInformatieObject on line {row_index}: \n"
+            f"{str(e)}"
+        )
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    file_path = document_row.bestandspad
+    path = Path(settings.IMPORT_DOCUMENTEN_BASE_DIR) / Path(file_path)
+
+    if not path.exists() or not path.is_file():
+        error_message = (
+            f"The given filepath {path} does not exist or is not a file for "
+            f"row {row_index}"
+        )
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    default_dir = get_default_path(EnkelvoudigInformatieObject.inhoud.field)
+    import_path = default_dir / path.name
+
+    if not default_dir.exists():
+        default_dir.mkdir(parents=True)
+
+    try:
+        shutil.copy2(path, import_path)
```

Comment:
I am curious how it file copying works with our docker setup. Should we create a special volume for it in our docker file? Have you tested it with docker-compose?

Reply:
Note that the current setup does work out of the box for any setup although all the import metadata files would have to be stored in the `BASE_DIR` (the default of `IMPORT_DOCUMENTEN_BASE_DIR`). Having a dedicated directory for imports does feel a bit more organized though.

Reply:
Let's put everything into one docker-compose file, at least when it's possible, so it would be similar to real deployment configuration